service/service.go

```diff
@@ -0,0 +1,224 @@
+package service
+
+import (
+  "fmt"
+  "net/http"
+  "strconv"
+  "sync"
+  "time"
+  "errors"
+
+  "github.com/sirupsen/logrus"
+  "github.com/halarcon-wizeline/academy-go-q12021/domain"
+  "github.com/halarcon-wizeline/academy-go-q12021/infrastructure/datastore"
+  "github.com/halarcon-wizeline/academy-go-q12021/infrastructure/externalapi"
+)
+
+// Service
+type Service struct {
+  logger  *logrus.Logger
+}
+
+// New creates a new Service layer
+func New(
+  logger  *logrus.Logger,
+  ) (*Service, error) {
+  return &Service{logger}, nil
+}
+
+// GetExternalPokemons logic
+func (s *Service) GetExternalPokemons() (string, error) {
+  fmt.Println("[controller] GetExternalPokemons")
+
+  logger := s.logger.WithFields(logrus.Fields{"func": "Get Pokemons Api", })
+  logger.Debug("in")
+
+  pokemons, err := externalapi.GetPokemons()
+  if err != nil {
+    logger.WithError(err).Error("Getting external api")
+    return "", errors.New("Can't find api")
+  }
+
+  file, err := datastore.CreatePokemonDB("./infrastructure/datastore/pokemons_api.csv", pokemons)
+  if err != nil {
+    logger.WithError(err).Error("Exporting csv file")
+    return "", errors.New("Export failed")
+  }
+  fmt.Println(file)
+
+  return file, nil
+}
+
+// GetLocalPokemons logic
+func (s *Service) GetLocalPokemons() ([]domain.Pokemon, error) {
+  fmt.Println("[controller] GetLocalPokemons")
+
+  logger := s.logger.WithFields(logrus.Fields{"func": "Get Local Pokemons", })
+  logger.Debug("in")
+
+  pokemons, err := datastore.GetPokemonDB()
+  if err != nil {
+    logger.WithError(err).Error("Getting local pokemons")
+    return []domain.Pokemon{}, errors.New("Pokemons db not found")
+  }
+
+  return pokemons, nil
+}
+
+// GetLocalPokemon logic
+func (s *Service) GetLocalPokemon(id string) (domain.Pokemon, error) {
+
+  var pokemon = domain.Pokemon{}
+  fmt.Println("[controller] GetLocalPokemon")
+
+  logger := s.logger.WithFields(logrus.Fields{"func": "Get Local Pokemon", })
+  logger.Debug("in")
+
+  pokemons, err := datastore.GetPokemonDB()
+  if err != nil {
+    logger.WithError(err).Error("Getting local pokemon")
+    return domain.Pokemon{}, errors.New("Pokemons db not found")
+  }
+
+  val, err := strconv.Atoi(id)
+  if err != nil {
+    logger.WithError(err).Error("Couldn't convert value")
+    return pokemon, errors.New("Couldn't convert value")
+  }
+
+  for _, poke := range pokemons {
+    if poke.ID == val {
+      return poke, nil
+    }
+  }
+  return pokemon, nil
+}
+
+// GetLocalPokemonWorkers logic
+func (s *Service) GetLocalPokemonWorkers(r *http.Request) ([]domain.Pokemon, error) {
+  fmt.Println("[controller] GetLocalPokemonWorkers")
+
+  var pokemons = []domain.Pokemon{}
+
+  // Default values
+  pItems := 10
```

Comment:
If you need to set default values, you can consider setting the values as constants and initializing the default values on the `func init()` and only here doing the overwriting if necessary